### PR TITLE
Feature/dsci 108 refactor for new inputs and outputs

### PR DESCRIPTION
**Pull request by kdutia**
Minimally functional embeddings generation and indexer. There's [loads more to do on this](https://linear.app/climate-policy-radar/project/search-indexer-for-ga-e82f17b24832), including some stuff that has dependencies on the loader work.

I'll finish updating the readme when i add Dockerfiles to this repo.

Outstanding questions:
* do we still need PDF md5sum in the index?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# Data folder
+data/**
+!data/**/.gitkeep
```
Comment (diversemix) [inline .gitignore:3]: ignoring a gitkeep seems bizarre - can we not just remove the gitkeep?

Comment (diversemix) [reply .gitignore:3]: silly me - missed the not

File: cli/test/test_text2embeddings.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+def test_run_encoder_local(test_input_dir: Path):
+    """Test that the encoder runs with local input and output paths and outputs the correct files."""
+
+    with tempfile.TemporaryDirectory() as output_dir:
+        runner = CliRunner()
+        result = runner.invoke(cli_main, [str(test_input_dir), output_dir])
+        assert result.exit_code == 0
+
+        assert (Path(output_dir) / "test_html.npy").exists()
```
Comment (diversemix) [inline cli/test/test_text2embeddings.py:26]: Might be worth checking these don't exist (and removing them) as part of the test setup?

File: cli/test/test_text2embeddings.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        runner = CliRunner()
+        result = runner.invoke(
+            cli_main,
+            [
+                str(test_input_dir),
+                output_dir,
+            ],
+        )
+
+        assert result.exit_code == 0
```
Comment (diversemix) [inline cli/test/test_text2embeddings.py:68]: For completeness - might be worth having a test that ensures the cli errors when bad data is passed

File: cli/test/test_text2embeddings.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+    with tempfile.TemporaryDirectory() as output_dir:
+        runner = CliRunner()
+        result = runner.invoke(cli_main, [str(test_input_dir), output_dir])
+        assert result.exit_code == 0
+
+        assert (Path(output_dir) / "test_html.npy").exists()
+        assert (Path(output_dir) / "test_html.json").exists()
+        assert (Path(output_dir) / "test_pdf.npy").exists()
+        assert (Path(output_dir) / "test_pdf.json").exists()
```
Comment (diversemix) [inline cli/test/test_text2embeddings.py:29]: Is there a test that validates this is in fact json and/or with the correct schema?

**Review by diversemix — APPROVED** [review]
One or two comment - but no showstoppers - nice refactor and good readable code :+1:

File: cli/test/test_text2embeddings.py (hunk 2 above)
Comment (kdutia) [reply cli/test/test_text2embeddings.py:26]: doesn't `TemporaryDirectory()` as a context manager remove the directory as soon as it's exited anyway?

Comment (diversemix) [reply cli/test/test_text2embeddings.py]: thanks for the education :smile:

**Pull request merged**